catalog/controller/extension/module/tawkto.php

```diff
@@ -101,33 +113,41 @@ private function getWidget() {
                 $show = false;
 
                 $current_page = (string) $current_page;
-                foreach ($show_pages as $slug) {
-                    $slug = trim($slug);
-                    if (empty($slug)) {
-                        continue;
-                    }
-
-                    // use this when testing on a Linux/Win
-                    // $slug = (string) htmlspecialchars($slug); // we need to add htmlspecialchars due to slashes added when saving to database
-                    $slug = (string) urldecode($slug); // we need to add htmlspecialchars due to slashes added when saving to database
-                    $slug = str_ireplace($this->config->get('config_url'), '', $slug);
 
-                    // use this when testing on a Mac
-                    // $slug = (string) urldecode($slug); // we need to add htmlspecialchars due to slashes added when saving to database
-
-                    $slug = addslashes($slug);
-
-                    // $slug = urlencode($slug);
-                    if (stripos($current_page, $slug)!==false || trim($slug)==trim($current_page)) {
+                // handle backwards compatibility
+                if (version_compare($plugin_version_in_db, PATTERN_MATCHING_UPDATE_VERSION) < 0) {
```

Comment:
Would recommend wrapping this in a standalone function (i.e. `shouldShowWidget`)
That way you can utilize early return and put pattern matching first so the smallest `if` (and the default path) comes first

Reply:
On a side-note : do we need this since we already have a migration script for upgrading?

Reply:
I added this because of how we're doing the db upgrade. Since db upgrades are only accessible on the admin pages, I added this for backwards compatibility just in case the upgrade script hasn't ran yet and users accessed their website.

Reply:
Done